### PR TITLE
enh(UI): Reduce Redux usage and rework poller wizard

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This reduces Redux usage in the application in order to use Jotai.
This refactors all the code related to the navigation data and external components.
This also rework a bit the poller wizard to use a single React route `/poller-wizzard/1`.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

- The poller wizard should work as before
- Modules pages / hooks should be accessible
- The menu should be correctly displayed

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/front_src/src/Header/userMenu/index.js [hunk 1]
```diff
@@ -226,12 +225,10 @@ class UserMenu extends Component {
   }
 }
 
-const mapStateToProps = (state) => ({
-  allowedPages: allowedPagesSelector(state),
-});
+const UserTopCounter = (props) => {
```
Comment (bdauria) [inline www/front_src/src/Header/userMenu/index.js:228]: This isn't really a TopCounter

File: www/front_src/src/Navigation/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { retrievedNavigation, allowedPages, reactRoutes } from './mocks';
+
+const mockedAxios = axios as jest.Mocked<typeof axios>;
+
+describe('navigation', () => {
+  beforeEach(() => {
+    mockedAxios.get.mockReset();
+    mockedAxios.get.mockResolvedValue({ data: retrievedNavigation });
+  });
+  it('gets the allowed pages with the navigation data retrieved from the API', async () => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/index.test.ts:15]:
```suggestion
  });
  
  it('gets the allowed pages with the navigation data retrieved from the API', async () => {
```

File: www/front_src/src/Navigation/index.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable no-undef */
+/* eslint-disable react/jsx-filename-extension */
+/* eslint-disable react/prop-types */
+/* eslint-disable no-shadow */
```
Comment (bdauria) [inline www/front_src/src/Navigation/index.tsx:4]: I think the reacto makes these comments deprecated

File: www/front_src/src/Navigation/useNavigation.ts [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  './api/internal.php?object=centreon_topology&action=navigationList';
+
+interface UseNavigationState {
+  allowedPages?: Array<string | Array<string>>;
+  getNavigation: () => void;
+  menu?: Array<Page>;
+  reactRoutes?: Record<string, string>;
+}
+
+const exists = pipe(isNil, not);
```
Comment (bdauria) [inline www/front_src/src/Navigation/useNavigation.ts:39]: I would name this isDefined instead, wdyt?

File: www/front_src/src/PollerWizard/forms/ServerConfigurationWizardForm.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import { useTranslation } from 'react-i18next';
+
+import { Button, Typography } from '@material-ui/core';
+
+import styles from '../../styles/partials/form/_form.scss';
+import RadioGroupFields from '../../components/form-fields/RadioGroupFields';
+
+const configurationTypes = [
+  {
+    label: 'Add a Centreon Remote Server',
```
Comment (jdelpierre) [inline www/front_src/src/PollerWizard/forms/ServerConfigurationWizardForm.js:17]: No translation here ?

File: www/front_src/src/PollerWizard/forms/ServerConfigurationWizardForm.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  return (
+    <div>
+      <div className={styles['form-heading']}>
+        <Typography variant="h6">{t('Choose a server type')}</Typography>
+      </div>
+      <form autoComplete="off" onSubmit={handleSubmit(onSubmit)}>
+        <Field
+          component={RadioGroupFields}
+          name="server_type"
+          options={configurationTypes}
```
Comment (jdelpierre) [inline www/front_src/src/PollerWizard/forms/ServerConfigurationWizardForm.js:38]: No translation here ?

**Review by jdelpierre — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**